### PR TITLE
[Debugger Expression Evaluation] Add support for rest arguments

### PR DESCRIPTION
**Pull request by NipunaRanasinghe**
## Purpose
This PR
- adds debugger expression evaluation support for function/object method/action invocations with rest arguments.
- adds integration test cases for related scenarios.
- Fixes https://github.com/ballerina-platform/ballerina-lang/issues/31719

## Check List 
- [x] Read the [Contributing Guide](https://github.com/ballerina-platform/ballerina-lang/blob/master/CONTRIBUTING.md)
- [ ] Updated Change Log
- [x] Checked Tooling Support (#<Issue Number>)
- [x] Added necessary tests
  - [ ] Unit Tests
  - [ ] Spec Conformance Tests
  - [x] Integration Tests
  - [ ] Ballerina By Example Tests
- [ ] Increased Test Coverage   
- [ ] Added necessary documentation  
  - [ ] API documentation 
  - [ ] Module documentation in Module.md files
  - [ ] Ballerina By Examples

### CODE REVIEW

File: misc/debug-adapter/modules/debug-adapter-core/src/main/java/org/ballerinalang/debugadapter/evaluation/engine/BallerinaTypeResolver.java [hunk 1]
```diff
@@ -73,7 +77,32 @@
         return resolvedTypes;
     }
 
+    /**
+     * @param context        debug context
+     * @param typeDescriptor type descriptor string
+     * @return a collection of resolved types
+     * @throws EvaluationException if unsupported type(s) found
```
Comment (prabushi) [inline misc/debug-adapter/modules/debug-adapter-core/src/main/java/org/ballerinalang/debugadapter/evaluation/engine/BallerinaTypeResolver.java:84]:
```suggestion
     * @throws {@link EvaluationException} if unsupported type(s) found
```

File: misc/debug-adapter/modules/debug-adapter-core/src/main/java/org/ballerinalang/debugadapter/evaluation/engine/InvocationArgProcessor.java [hunk 2]
```diff
@@ -72,34 +93,58 @@
...
 
-                String parameterName = params.get(i).getName().get();
-                Value argValue = arg.getValue().evaluate().getJdiValue();
-                // For the ballerina function parameters with type "any", the generated runtime method will only accept
-                // the args which are subtypes of "java.lang.Object". Therefore all the primitive typed arguments
-                // must be converted into their wrapper implementations (i.e. 'int' -> 'Integer'), before passing
-                // into the method.
-                if (params.get(i).typeDescriptor().typeKind().getName().equals(BVariableType.ANY.getString())) {
-                    argValue = getValueAsObject(context, argValue);
+                String parameterName = params.get(paramIndex).getName().get();
```
Comment (prabushi) [inline misc/debug-adapter/modules/debug-adapter-core/src/main/java/org/ballerinalang/debugadapter/evaluation/engine/InvocationArgProcessor.java:110]: Can we make parameterName a local variable inside the else block? #L144

Comment (NipunaRanasinghe) [reply misc/debug-adapter/modules/debug-adapter-core/src/main/java/org/ballerinalang/debugadapter/evaluation/engine/InvocationArgProcessor.java]: @prabushi changed in f131f53

File: misc/debug-adapter/modules/debug-adapter-core/src/main/java/org/ballerinalang/debugadapter/evaluation/engine/BallerinaTypeResolver.java (hunk 1 above)
Comment (NipunaRanasinghe) [reply misc/debug-adapter/modules/debug-adapter-core/src/main/java/org/ballerinalang/debugadapter/evaluation/engine/BallerinaTypeResolver.java:84]: @prabushi We can directly define Exception without the @link annotation as suggested in the [Oracle Javadocs documentation](https://www.oracle.com/technical-resources/articles/java/javadoc-tool.html#throwstag) (Also the intellij gives a warning when using `@link` with `@throws`).

File: misc/debug-adapter/modules/debug-adapter-core/src/main/java/org/ballerinalang/debugadapter/evaluation/engine/InvocationArgProcessor.java [hunk 3]
```diff
@@ -72,34 +93,58 @@
...
+                    Value elementType = getElementType(context, restArrayType);
+                    if (!checkIsType(context, jdiArgValue, elementType)) {
+                        throw new EvaluationException(String.format(EvaluationExceptionKind.TYPE_MISMATCH.getString(),
+                                restParamTypeName.replaceAll(BallerinaTypeResolver.ARRAY_TYPE_SUFFIX, ""),
+                                argValue.getType().getString(), restParamName));
+                    }
+                    restValues.add(jdiArgValue);
+                    remainingParams.remove(restParamName);
+                } else {
+                    String parameterName = params.get(paramIndex).getName().get();
```
Comment (praveennadarajah) [inline misc/debug-adapter/modules/debug-adapter-core/src/main/java/org/ballerinalang/debugadapter/evaluation/engine/InvocationArgProcessor.java:143]: Using `Optional.get()` without `isPresent()` check. Shall we add the necessary checks here?

Comment (NipunaRanasinghe) [reply misc/debug-adapter/modules/debug-adapter-core/src/main/java/org/ballerinalang/debugadapter/evaluation/engine/InvocationArgProcessor.java]: Fixed in f0ae64f

**Review by praveennadarajah — APPROVED** [review]

**Pull request merged**